core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -291,11 +291,20 @@ private[spark] object TaskMetrics extends Logging {
 
 private[spark] class BlockStatusesAccumulator
   extends AccumulatorV2[(BlockId, BlockStatus), Seq[(BlockId, BlockStatus)]] {
-  private[this] var _seq = ArrayBuffer.empty[(BlockId, BlockStatus)]
+  private var _seq = ArrayBuffer.empty[(BlockId, BlockStatus)]
 
   override def isZero(): Boolean = _seq.isEmpty
 
-  override def copyAndReset(): BlockStatusesAccumulator = new BlockStatusesAccumulator
+  override def copy(): BlockStatusesAccumulator = {
```

Comment:
We can just have a copyAndReset with default implementation to call copy and then reset. Specific accumulator implementations (like this one) can override that to optimize. Otherwise users have to implement too many functions.